### PR TITLE
Fix - Uploading Mp4 Videos Removes 'Attachment' Tag 

### PR DESCRIPTION
**Pull request by dennismunene**
### Fixed Issues

$ https://github.com/Expensify/App/issues/8331

### Tests

1. Open Expensify App and log in with any account
2. Click to add attachment
3. Select mp4 video and tap Send
4. Refresh the page

#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose and it is
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn’t already exist
    - [x] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

<details>
<summary><h4>PR Reviewer Checklist</h4></summary>

- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [] I verified tests pass on **all platforms** & I tested again on:
    - [] iOS / native
    - [] Android / native
    - [] iOS / Safari
    - [] Android / Chrome
    - [] MacOS / Chrome
    - [] MacOS / Desktop
- [] I verified there are no console errors (if there’s a console error not related to the PR, report it or open an issue for it to be fixed)
- [] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained “why” the code was doing something instead of only explaining “what” the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named “index.js”. All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn’t already exist
    - [ ] The style can’t be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.

</details>

### QA Steps

##Expected Result
- `[Attachment]` Tag should be shown for all attachment types. 
-  Links should not be shown as attachments 

LHN - Left Hand Navigation 

To test these changes on all platforms please follow these steps 

Testing File uploads (Images,PDF,Audio,Video) 

1. Open Expensify App and log in with any account
2. Click to add attachment
3. Select any to upload File e.g (Mp4 video) and tap Send 
4. This should show up on the LHN as `[Attachment]` instead of showing the filename
5. Refresh the page (on mobile go back to chats page)
6. Attachments should still show as `[Attachment]` on the LHN

Testing Links 

1. Open Expensify App and log in with any account
2. Navigate to any chat 
3. Copy paste/ Type in any link e.g https://google.com in the textbox and hit send
4. This should show up on the LHN as a link `https://google.com` and not as an Attachment
5. Refresh the page (on mobile go back to chats page)
6. Links should still show as links on the LHN

This should work on all platforms & for all attachments. 

- [ ] Verify that attachment tag is shown for all file uploads even after page refresh 
- [ ] Verify that links are not shown as attachments 
- [ ] Verify that no errors appear in the JS console

#### Web

https://user-images.githubusercontent.com/4717301/161214022-7fdaac4f-7435-4c91-9c55-84ab164bb8bd.mov

#### Mobile Web

https://user-images.githubusercontent.com/4717301/161274783-9154781a-bc42-468f-87da-34a09e88e5bf.mov

#### Desktop

https://user-images.githubusercontent.com/4717301/161274917-d175d078-7913-4ef5-9309-0be428faf701.mov

#### iOS

https://user-images.githubusercontent.com/4717301/161274944-133a58c8-9fb0-4a64-ad86-5e769fd3fcae.mov

#### Android

https://user-images.githubusercontent.com/4717301/161275076-a1349eeb-d2d7-47e2-9d72-b9dea93476da.mov

### CODE REVIEW

**Comment by parasharrajat** [comment]
Please correct the commit errors and all commits need to sign. Follow Contributor (PR Author) Checklist for code changes.

**Comment by dennismunene** [comment]
@parasharrajat fixed commit errors

**Review by marcaaron — CHANGES_REQUESTED** [review]
I think we missed what the original issue is asking for here.

**Review by parasharrajat — CHANGES_REQUESTED** [review]
Missing QA steps.

**Review by mountiny — COMMENTED** [review]
I _think_ the expected results should be to always show `[Attachment]` for any type of upload to unify it for now and remove the inconsistency. I think we understood each other despite the misleading issue expected results section somehow 😄

**Review by mountiny — COMMENTED** [review]
@dennismunene Can you please try to comment this again? Thank you
`I have read the CLA Document and I hereby sign the CLA`

**Review by parasharrajat — DISMISSED** [review]
LGTM. 

cc: @mountiny 

#### PR Reviewer Checklist
- [x] I verified the PR has a small number of commits behind `main`
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the testing environment is mentioned in the test steps
- [x] I verified testing steps cover success & fail scenarios (if applicable)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [ ] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors related to changes in this PR
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified comments were added when the code was not self explanatory
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files (if applicable)
    - [x] I verified proper naming convention for platform-specific files was followed (if applicable)
    - [x] I verified [style guidelines](https://github.com/Expensify/App/blob/main/STYLING.md) were followed
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components are not impacted by changes in this PR (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified the UI performance was not affected (the performance is the same than `main` branch)
- [x] If a new component is created I verified that a similar component doesn't exist in the codebase

:ribbon: :eyes: :ribbon:  C+ reviewed

**Comment by parasharrajat** [comment]
Please improve QA steps https://github.com/Expensify/App/pull/8438#issuecomment-1089059572

**Comment by marcaaron** [comment]
>I think we understood each other despite the misleading issue expected results section somehow 😄

Let's update the original issue then?

**Comment by marcaaron** [comment]
Oh sorry I saw you did. Thanks!

File: src/libs/actions/Report.js
Comment (marcaaron) [inline src/libs/actions/Report.js:186]: **suggestion:** add a constant for `data-expensify-source`. There are a few usages in the codebase and it the code would be clearer if we used a constant that said what `data-expensify-source` is vs. directly reference it.

something like `CONST.ATTACHMENT_ATTRIBUTE_SOURCE` maybe?

**Review by marcaaron — DISMISSED** [review]

**Comment by dennismunene** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/libs/actions/Report.js
Comment (mountiny) [reply src/libs/actions/Report.js:186]: @dennismunene Let us know if you need any help with this. 

Great suggestion!

Comment (dennismunene) [reply src/libs/actions/Report.js:186]: @mountiny Yes would appreciate some help with this ,  do i need to raise a PR here https://github.com/Expensify/expensify-common to introduce the new CONST ?

Comment (mountiny) [reply src/libs/actions/Report.js:186]: @dennismunene Nono, I think adding the const [here](https://github.com/Expensify/App/blob/ee5d18e2bb47a9d2b71fcca08ae1254b56a95d4c/src/CONST.js#L11-L657) and then importing it _should_ be sufficient :) 

We should then look into a follow up to refactor the other places where there is a use of `data-expensify-source` to update it, but we do not have to do it in this PR!

Comment (dennismunene) [reply src/libs/actions/Report.js:186]: Great , thank you 🙂

Comment (dennismunene) [reply src/libs/actions/Report.js]: Pushed update with this suggestion. 

What is the correct flow to follow up on the refactor task after this PR is merged ?

**Review by mountiny — APPROVED** [review]
LGTM! Congratulations to the first PR @dennismunene! Great job on this one!

Thank you @parasharrajat for the review!

**Comment by mountiny** [comment]
@dennismunene In terms of the refactor, there are not many occurrences of the `data-expensify-source` in the codebase, so would you be able to create a simple PR that will substitute them for the `CONST` and link it to the original issue here? 

Thank you very much for your work here, I hope to see many more PRs from you since the work you have done here has been very professional 🙇 🙌

**Comment by mountiny** [comment]
@marcaaron Will you be able to approve and merge if you are happy with this one? Thank you very much! (dont want to dismiss your request for changes 😄 )

**Comment by dennismunene** [comment]
@mountiny Thank you. I've had a great experience working on this ,  I will definitely be raising more PR's in future 🙂

**Comment by mountiny** [comment]
I have been mean and went ahead and dismissed the review as the requested change has been addressed! Thank you very much for chiming in @marcaaron, @dennismunene will follow up with a refactor to address the other mentions of `data-expensify-source`!

**Comment by mountiny** [comment]
@dennismunene Very glad to hear this 🙌

**Comment by dennismunene** [comment]
> I will be mean and go ahead and dismiss the review as the requested change has been addressed! Thank you very much for chiming in @marcaaron, @dennismunene will follow up with a refactor to address the other mentions of `data-expensify-source`!

@mountiny @marcaaron  Added the Refactor PR here https://github.com/Expensify/App/pull/8539

**Pull request merged**
